lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,426 @@
+						v-if="headerFeatureIcon"
+						class="ds-modal__headerFeatureIcon"
+						:color="calcHeaderFeatureIconColor"
+						:icon="headerFeatureIcon"
+						:size="FEATURE_ICON_SIZES.X_LARGE"
+						double-background
+					/>
+					<h4
+						class="ds-modal__headerTitle"
+						:class="{ '-small': headerTitleSize === MODAL_HEADER_TITLE_SIZES.SMALL }"
+						>{{ headerTitle }}</h4
+					>
+					<h5 v-if="headerSubtitle" class="ds-modal__headerSubtitle">{{
+						headerSubtitle
+					}}</h5>
+				</div>
+				<div v-if="$slots.default" class="ds-modal__slotContent">
+					<slot />
+				</div>
+				<div class="ds-modal__footer" :class="{ '-singleColumn': calcSingleColumn }">
+					<div
+						v-if="footerTertiaryButtonText || footerCheckboxText"
+						class="ds-modal__footerColumn --ctaSecondary"
+					>
+						<div v-if="footerCheckboxText" class="ds-modal__checkbox">
+							<input
+								id="ds-modal__checkboxInput"
+								type="checkbox"
+								:checked="false"
+								class="ds-modal__checkboxInput"
+                @change="$emit('checkbox-change', $event.target.checked)"
+							/>
+							<label for="ds-modal__checkboxInput" class="ds-modal__checkboxLabel">
+								{{ footerCheckboxText }}
+							</label>
+						</div>
+						<wnl-button
+							v-if="footerTertiaryButtonText"
+							ref="button"
+							class="ds-modal__buttonSecondary"
+							:type="BUTTON_TYPES.TEXT"
+							:color="BUTTON_COLORS.NEUTRAL"
+							:icon-left="ICONS[footerTertiaryButtonIcon]"
+						>
+							{{ footerTertiaryButtonText }}
+						</wnl-button>
+					</div>
+					<div
+						v-if="footerSecondaryButtonText || footerPrimaryButtonText"
+						class="ds-modal__footerColumn --cta"
+					>
+						<wnl-button
+							v-if="footerSecondaryButtonText"
+							ref="button"
+							class="ds-modal__buttonSecondary"
+							:type="BUTTON_TYPES.OUTLINED"
+							:color="calcFooterSecondaryButtonColor"
+							:icon-right="ICONS[footerSecondaryButtonIcon]"
+						>
+							{{ footerSecondaryButtonText }}
+						</wnl-button>
+						<wnl-button
+							v-if="footerPrimaryButtonText"
+							ref="button"
+							class="ds-modal__buttonPrimary"
+							:color="calcFooterPrimaryButtonColor"
+							:icon-right="ICONS[footerPrimaryButtonIcon]"
+						>
+							{{ footerPrimaryButtonText }}
+						</wnl-button>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/radiuses';
+@import '../../../../styles/settings/shadows';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/typography/tokens';
+@import '../../../../styles/settings/icons';
+@import '../../../../styles/settings/media-queries';
+
+$modal-medium-width: 700px;
+$modal-small-width: 460px;
+$image-height: 200px;
+$image-height-small: 140px;
+
+.ds-modal {
+	$self: &;
+
+  background: $color-default-overlay;
+  height: 100%;
```

Comment:
IMO, we should use `position: fixed` to block scrolling content under the Modal. Then setting `height:100%` probably won't work

Reply:
Added.